File: apps/portal/web/src/api/kvcache.ts
import { apiFetch, buildQuery, type ListParams, type Pagination } from "./client";

export interface KVCacheListItem {
  name: string;
  namespace: string;
  mode: string;
  readyReplicas: number;
  createdAt: string;
}

export interface KVCacheListResponse {
  items: KVCacheListItem[];
  pagination: Pagination;
}

export interface KVCacheCreateRequest {
  name: string;
  namespace: string;
  mode?: string;
}

export interface KVCacheDetail {
  name: string;
  namespace: string;
  mode: string;
  status: { readyReplicas: number };
  createdAt: string;
}

export const kvCacheApi = {
  list: (params?: ListParams) =>
    apiFetch<KVCacheListResponse>(`/kvcaches${buildQuery(params)}`),
  get: (ns: string, name: string) =>
    apiFetch<KVCacheDetail>(`/kvcaches/${ns}/${name}`),
  create: (data: KVCacheCreateRequest) =>
    apiFetch<KVCacheDetail>("/kvcaches", { method: "POST", body: JSON.stringify(data) }),
  delete: (ns: string, name: string) =>
    apiFetch<void>(`/kvcaches/${ns}/${name}`, { method: "DELETE" }),
};

File: pkg/portal/types/kvcache.go
package types

import "time"

type KVCacheCreateRequest struct {
	Name      string `json:"name" binding:"required"`
	Namespace string `json:"namespace" binding:"required"`
	Mode      string `json:"mode,omitempty"`
}

type KVCacheUpdateRequest struct {
	Mode string `json:"mode,omitempty"`
}

type KVCacheListItem struct {
	Name          string    `json:"name"`
	Namespace     string    `json:"namespace"`
	Mode          string    `json:"mode"`
	ReadyReplicas int32     `json:"readyReplicas"`
	CreatedAt     time.Time `json:"createdAt"`
}

type KVCacheListResponse struct {
	Items      []KVCacheListItem `json:"items"`
	Pagination Pagination        `json:"pagination"`
}

type KVCacheDetailResponse struct {
	Name      string              `json:"name"`
	Namespace string              `json:"namespace"`
	Mode      string              `json:"mode"`
	Status    KVCacheDetailStatus `json:"status"`
	CreatedAt time.Time           `json:"createdAt"`
}

type KVCacheDetailStatus struct {
	ReadyReplicas int32 `json:"readyReplicas"`
}
